editor/src/messages/tool/tool_messages/select_tool.rs

```diff
@@ -610,30 +860,35 @@ impl Fsm for SelectToolFsmState {
 					if !tool_data.is_dragging && input.keyboard.get(remove_from_selection as usize) && tool_data.layer_selected_on_start.is_none() {
 						let quad = tool_data.selection_quad();
 						let intersection = document.document_legacy.intersects_quad_root(quad, render_data);
-						if let Some(intersect_layer_path) = intersection.last() {
-							let replacement_selected_layers = document.selected_layers().filter(|&layer| layer != intersect_layer_path).map(|path| path.to_vec()).collect();
-							responses.push_back(DocumentMessage::SetSelectedLayers { replacement_selected_layers }.into());
-						}
+						let temp_intersection = intersection.clone();
+
+						let path: &[u64] = temp_intersection.last().unwrap();
+						let folders: Vec<_> = (1..path.len() + 1).map(|i| &path[0..i]).collect();
+						let replacement_selected_layers: Vec<Vec<u64>> = document.selected_layers().filter(|&layer| !folders.contains(&layer)).map(|path| path.to_vec()).collect();
+
+						tool_data.layers_dragging.clear();
+						tool_data.layers_dragging.append(&mut replacement_selected_layers.clone());
+
+						responses.push_back(DocumentMessage::SetSelectedLayers { replacement_selected_layers }.into());
 					}
 
 					tool_data.is_dragging = false;
 					tool_data.layer_selected_on_start = None;
 
-					let response = match input.mouse.position.distance(tool_data.drag_start) < 10. * f64::EPSILON {
-						true => DocumentMessage::Undo,
-						false => DocumentMessage::CommitTransaction,
-					};
+					// let _response = match input.mouse.position.distance(tool_data.drag_start) < 10. * f64::EPSILON {
+					// 	true => DocumentMessage::Undo,
+					// 	false => DocumentMessage::CommitTransaction,
+					// };
+					responses.push_back(DocumentMessage::CommitTransaction.into());
 					tool_data.snap_manager.cleanup(responses);
-					responses.push_front(response.into());
-
 					Ready
 				}
 				(ResizingBounds, DragStop { .. } | Enter) => {
-					let response = match input.mouse.position.distance(tool_data.drag_start) < 10. * f64::EPSILON {
+					let _response = match input.mouse.position.distance(tool_data.drag_start) < 10. * f64::EPSILON {
```

Comment:
An underscore before an identifier name usually implies that it is unused - perhaps the underscore could be removed?